2021/17.py

```diff
@@ -27,9 +36,10 @@
             elif dx < 0:
                 dx += 1
             dy -= 1
-            #if 20<=x<=30 and -10<=y<=-5:
-            if 96<=x<=125 and -144<=y<=-98:
+
+            if MinX <= x <= MaxX and MinY <= y <= MaxY:
                 ok = True
+                break
```

Comment:
Once the probe is in the target range you can break out of the loop as a small optimization.